cli/src/lib.rs

```diff
@@ -4742,18 +4765,54 @@ fn get_node_version() -> Result<Version> {
     Version::parse(output).map_err(Into::into)
 }
 
-fn get_recommended_micro_lamport_fee(client: &RpcClient, priority_fee: Option<u64>) -> Result<u64> {
-    if let Some(priority_fee) = priority_fee {
-        return Ok(priority_fee);
+fn add_recommended_deployment_solana_args(
+    client: &RpcClient,
+    args: Vec<String>,
+) -> Result<Vec<String>> {
+    let mut augmented_args = args.clone();
+
+    // If no priority fee is provided, calculate a recommended fee based on recent txs.
+    if !args.contains(&"--with-compute-unit-price".to_string()) {
+        let priority_fee = get_recommended_micro_lamport_fee(&client)?;
```

Comment:
This results in `clippy::needless_borrow`: https://github.com/coral-xyz/anchor/actions/runs/11765313625/job/32872631172?pr=3354#step:8:549

```
error: this expression creates a reference which is immediately dereferenced by the compiler
    --> cli/src/lib.rs:4776:62
     |
4776 |         let priority_fee = get_recommended_micro_lamport_fee(&client)?;
     |                                                              ^^^^^^^ help: change this to: `client`
     |
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrow
     = note: `-D clippy::needless-borrow` implied by `-D warnings`
     = help: to override `-D warnings` add `#[allow(clippy::needless_borrow)]`
```

```suggestion
        let priority_fee = get_recommended_micro_lamport_fee(client)?;
```